springwolf-core/src/main/java/io/github/stavshamir/springwolf/asyncapi/MessageHelper.java

```diff
@@ -13,6 +19,10 @@ public class MessageHelper {
     private static final String ONE_OF = "oneOf";
 
     private static final Comparator<Message> byMessageName = Comparator.comparing(Message::getName);
+
+    // TODO Why do we need a SortedSet here? Using a comparator with only the message name will break deep equals on the Set
```

Comment:
I'm not sure of having a sorted set here is a requirement, but using the TreeSet for Sorting and deduplication causes issues when comparing (ie test do not break). Maybe there is a better solution than just using a hashset (looses ordering), but for the tests it seems to be ok ...